snuba/subscriptions/partitioner.py

```diff
@@ -11,18 +11,15 @@ def build_partition_id(self, data: SubscriptionData) -> PartitionId:
         pass
 
 
-class DatasetSubscriptionDataPartitioner(SubscriptionDataPartitioner):
+class TopicSubscriptionDataPartitioner(SubscriptionDataPartitioner):
     """
-    Partitions a subscription based on the Dataset that we're going to store it in.
+    Identifies the partition index that contains the source data for a subscription.
     """
 
-    PARTITION_COUNT = 64
-
-    def __init__(self, dataset: Dataset):
-        self.__dataset = dataset
+    def __init__(self, topic: KafkaTopicSpec):
+        self.__topic = topic
 
     def build_partition_id(self, data: SubscriptionData) -> PartitionId:
```

Comment:
In retrospect, this maybe should have just returned a `Partition` object instead of the partition index. I'd like to make that change later but I don't think it's important enough to do right now.